plugins/module_utils/foreman_helper.py

```diff
@@ -820,6 +820,36 @@ def find_network(self, name, compute_resource, cluster=None):
             self.fail_json(msg=err_msg)
         return network
 
+    def find_storage_domain(self, name, compute_resource, cluster=None):
```

Comment:
at this point, the four find methods look really similar, so probably should be dried up.